pkg/controller/operators/openshift/clusteroperator_controller_test.go

```diff
@@ -245,5 +275,54 @@ var _ = Describe("ClusterOperator controller", func() {
 			}.String(),
 			LastTransitionTime: fixedNow(),
 		}))
+
+		By("setting upgradeable=false when invalid max versions are found")
+		incompatible.SetAnnotations(withMax(`"garbage"`))
+
+		Eventually(func() error {
+			return k8sClient.Update(ctx, incompatible)
+		}).Should(Succeed())
+
+		_, parseErr := semver.ParseTolerant("garbage")
+		Eventually(func() ([]configv1.ClusterOperatorStatusCondition, error) {
+			err := k8sClient.Get(ctx, clusterOperatorName, co)
+			return co.Status.Conditions, err
+		}, timeout).Should(ContainElement(configv1.ClusterOperatorStatusCondition{
+			Type:   configv1.OperatorUpgradeable,
+			Status: configv1.ConditionFalse,
+			Reason: IncompatibleOperatorsInstalled,
+			Message: skews{
+				{
+					namespace: ns.GetName(),
+					name:      incompatible.GetName(),
+					err:       fmt.Errorf(`Failed to parse "garbage" as semver: %w`, parseErr),
+				},
+			}.String(),
+			LastTransitionTime: fixedNow(),
+		}))
+
+		By("setting upgradeable=false when more than one max version property is defined")
+		incompatible.SetAnnotations(withMax(fmt.Sprintf(`"%s"`, clusterVersion), fmt.Sprintf(`"%s"`, next.String())))
+
+		Eventually(func() error {
+			return k8sClient.Update(ctx, incompatible)
+		}).Should(Succeed())
+
+		Eventually(func() ([]configv1.ClusterOperatorStatusCondition, error) {
+			err := k8sClient.Get(ctx, clusterOperatorName, co)
+			return co.Status.Conditions, err
+		}, timeout).Should(ContainElement(configv1.ClusterOperatorStatusCondition{
+			Type:   configv1.OperatorUpgradeable,
+			Status: configv1.ConditionFalse,
+			Reason: IncompatibleOperatorsInstalled,
+			Message: skews{
+				{
+					namespace: ns.GetName(),
+					name:      incompatible.GetName(),
+					err:       fmt.Errorf(`Defining more than one "%s" property is not allowed`, MaxOpenShiftVersionProperty),
```

Comment:
For future reference, you can also use `%q` instead of "%s" to get a quoted string, which helps when you'd rather use double quotes to define the string literal.